src/components/Tooltip/index.js

```diff
@@ -0,0 +1,202 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './style.css';
+
+import Icon from '../Icon';
+
+export const directions = ['top', 'left', 'right', 'bottom'];
+
+const Tooltip = ({
+  children,
+  closeable,
+  closeOnEscape,
+  content: tooltipContent,
+  direction,
+  isOpen,
+  showOnHover,
+  tooltipProps: tooltipOptsProp,
+  tooltipWidth,
+  ...rest
+}) => {
+  const [tooltipId] = useState(
+    `tooltip-${Date.now() * Math.floor(100 * Math.random())}`
+  );
+  const tooltipRef = useRef(null);
+
+  const [hovered, setHovered] = useState(false);
+  const [tooltipHeight, setTooltipHeight] = useState(0);
+
+  const closeFunc = useCallback(closeable || function() {}, []);
+
+  const handleSetHover = value => () => {
+    if (showOnHover) {
+      setHovered(value);
+    }
+  };
+
+  const handleEscape = e => {
```

Comment:
Since this is only being used within the following `useEffect` why not moving it to the if block inside?
```typescript
  useEffect(() => {
    if (closeOnEscape) {
      const handleEscape = e => {
        if (closeable && e.key === 'Escape') {
          closeable();
        } 
      };

      document.addEventListener('keydown', handleEscape);
      return () => {
        document.removeEventListener('keydown', handleEscape);
      };
    }

    return () => {};
  }, []);
```

Reply:
Good point. 👍